enzy_htp/structure/structure_operation/general.py

```diff
@@ -126,4 +128,19 @@ def align_atom_order_in_each_residue(stru: Structure, ref_stru: Structure) -> No
             atom_indexes = res.atom_idx_list
             atom_indexes.sort()
             res.renumber_atoms(atom_indexes)
-    return stru
\ No newline at end of file
+    return stru
+
+
+def update_coordinates(stru:Structure, ref_stru:Structure, allow_index_mismatch:bool=True) -> None:
```

Comment:
We have apply_geom already in Structure. It also check the topology to avoid potential hard to find bugs.

Reply:
Yeah this is a function I made that I forgot about. Basically I was finding a lot of errors because when `update_residues` is done it creates a deepcopy and will decouple `StructureConstraint`. I have since discovered a fix and do not use this.

Reply:
Cool! We can remove it in this case

Reply:
It has been removed! Sorry for the confusion